src/e2e/java/teammates/e2e/pageobjects/InstructorFeedbackResultsPage.java

```diff
@@ -0,0 +1,1144 @@
+
+import teammates.common.datatransfer.FeedbackParticipantType;
+import teammates.common.datatransfer.attributes.FeedbackQuestionAttributes;
+import teammates.common.datatransfer.attributes.FeedbackResponseAttributes;
+import teammates.common.datatransfer.attributes.FeedbackResponseCommentAttributes;
+import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.datatransfer.questions.FeedbackConstantSumQuestionDetails;
+import teammates.common.datatransfer.questions.FeedbackConstantSumResponseDetails;
+import teammates.common.datatransfer.questions.FeedbackContributionResponseDetails;
+import teammates.common.datatransfer.questions.FeedbackMcqQuestionDetails;
+import teammates.common.datatransfer.questions.FeedbackRankOptionsQuestionDetails;
+import teammates.common.datatransfer.questions.FeedbackRankOptionsResponseDetails;
+import teammates.common.datatransfer.questions.FeedbackResponseDetails;
+import teammates.common.datatransfer.questions.FeedbackRubricQuestionDetails;
+import teammates.common.datatransfer.questions.FeedbackRubricResponseDetails;
+import teammates.common.util.TimeHelper;
+
+/**
+ * Represents the "Results" page for Instructors.
+ */
+public class InstructorFeedbackResultsPage extends AppPage {
+    private static final String QUESTION_VIEW = "QUESTION";
+    private static final String GQR_VIEW = "GQR";
+    private static final String RQG_VIEW = "RQG";
+    private static final String GRQ_VIEW = "GRQ";
+    private static final String RGQ_VIEW = "RGQ";
+
+    private static final String NO_RESPONSE_LABEL = "No Response";
+    private static final String NO_TEAM_LABEL = "No Specific Team";
+    private static final String NO_SECTION_LABEL = "No specific section";
+    private static final String NO_USER_LABEL = "No Specific User";
+
+    private static final String MCQ_OTHER = "Other";
+
+    private String currentView = "";
+
+    @FindBy(id = "course-id")
+    private WebElement courseId;
+
+    @FindBy(id = "session-name")
+    private WebElement sessionName;
+
+    @FindBy(id = "session-duration")
+    private WebElement sessionDuration;
+
+    @FindBy(id = "result-visible-date")
+    private WebElement resultVisibleDate;
+
+    @FindBy(id = "btn-publish")
+    private WebElement publishButton;
+
+    @FindBy(id = "btn-download")
+    private WebElement downloadButton;
+
+    @FindBy(id = "no-response-panel")
+    private WebElement noResponsePanel;
+
+    @FindBy(id = "btn-remind-all")
+    private WebElement remindAllButton;
+
+    @FindBy(id = "include-team-grouping")
+    private WebElement groupByTeamCheckbox;
+
+    @FindBy(id = "include-statistics")
+    private WebElement statisticsCheckbox;
+
+    @FindBy(id = "include-missing-responses")
+    private WebElement missingResponsesCheckbox;
+
+    @FindBy(id = "btn-expand-all")
```

Comment:
This button only appears after some data is loaded, so adding this hook at the beginning may result in `NoSuchElementException` especially in production server